tests/coverage/test_coverage_multiprocessing.py

```diff
@@ -1,6 +1,101 @@
 import pytest
 
 
+@pytest.mark.subprocess(parametrize={"start_method": ["fork", "forkserver", "spawn"]})
+def test_coverage_multiprocessing_without_coverage():
+    """This test ensures that the coverage collector does not interfere with multiprocessing when it is not enabled."""
+    import multiprocessing
+
+    if __name__ == "__main__":
```

Comment:
Why do we have this in tests?

Reply:
The `subprocess` pytest mark causes the code to run in a new subprocess, so the function's essentially the whole program, which means using that, without `if __name__ == "main"` , the multiprocessing call happens at the module level, and gets you into the situation where:
```
RuntimeError:
        An attempt has been made to start a new process before the
        current process has finished its bootstrapping phase.
```

Technically, `free_support()` can be omitted, so I guess I can drop those.

Reply:
Oh and the tests run in a subprocess primarily because it just gives more predictability over when the `ModuleCodeCollector` is installed as well as what's happening in-process when tests are running, so things should be more deterministic (and less flaky) than trying to run tests within the `pytest` process.